lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3Context.java

```diff
@@ -49,7 +49,7 @@ public RequestPayer requestPayer()
 
     public S3Context withKmsKeyId(String kmsKeyId)
     {
-        return new S3Context(partSize, requesterPays, sseType, kmsKeyId, credentialsProviderOverride, cannedAcl, exclusiveWriteSupported);
```

Comment:
Let's verify it properly

Reply:
But it would have a restriction on S3SecurityMapping side that all the buckets should be under this KMS - For instance we could have one bucket which has `S3` security mapping and we could have one bucket which could have `KMS` type SSE which could be mapped to specific users - If we add a verification they we might not be able access bucket with different SSE settings. Or am I missing something here ?

Reply:
Isn't the filesystem created per bucket?

Reply:
In case of S3SecurityMapping or in general ? But the configs are shared right ?

Reply:
In general, the `TrinoFileSystemFactory` is created once at injection time and then `TrinoFileSystem` is created at each usage site with `ConnectorIdentity`. 

S3 security mapping uses `SwitchingFileSystemFactory` with `S3FileSystemLoader` to create a separate `S3SecurityMappingFileSystemFactory` per location.

Reply:
This change is correct. The purpose of S3 security mapping is to override the security settings for that mapping. So you could configure `s3.sse.type=S3` as the global default and override for a specific bucket, path, etc.